EIPS/eip-X-ASIC-resistant-block-mining.md

```diff
@@ -0,0 +1,182 @@
+
+
+## Abstract
+
+There are companies who currently have dedicated hardware based ethereum miners in
+production, and may be actively mining.  This EIP aims to "poison
+the well" by modifying the block mining algorithm in a low risk manner that
+may *"break"* these miners if they are in-fact built as traditional ASICs.
+
+
+## Motivation
+
+ASIC based miners will have lower operational costs than GPU based miners which
+will result in GPU based mining quickly becoming unprofitable.  Given that
+production of ASIC based miners has a high barrier to entry and few market players,
+this will cause a trend towards centralization of mining power.  
+
+This trend towards centralization has a negative effect on network security,
+putting significant control of the network in the hands of only a few entities.
+
+Ethash remains ASIC resistant, however ASIC manufacturer technology is advancing
+and ethash may require further changes in order to remain resistant to unforeseen
+design techniques. This EIP seeks explicitly to buy time during which newly developed
+ASIC technology will face a barrier while more long term mechanisms to ensure 
+continued ASIC resistance can be explored.  
+
+## Specification
+
+If `block.number >= ASIC_MITIGATION_FORK_BLKNUM`, require that the ethash solution 
+sealing the block has been mined using `ethashV2`.
+
+## EthashV2
+
+`ethashV2` will be identical in specification to the current `ethash`(v1) algorithm, with
+the exception of the implementation of `fnv`. 
+
+The new algorithm replaces the 5 current uses of `fnv` inside `hashimoto` with 5 
+seperate instances defined as `fnvA`, `fnvB`, `fnvC`, `fnvD`, and `fnvE`, utilizing 
+
+`FNV_PRIME_A=0x10001a7`  
+`FNV_PRIME_B=0x10001ab`  
+`FNV_PRIME_C=0x10001cf`  
+`FNV_PRIME_D=0x10001e3`  
+`FNV_PRIME_E=0x10001f9`  
+
+
+`fnvA` replaces `fnv` in the DAG item selection step
+`fnvB` replaces `fnv` in the DAG item mix step
+`fnvC(fnvD(fnvE` replaces `fnv(fnv(fnv(` in the compress mix step
+
+`fnv` as utilized in DAG-item creation should remain unchanged.
+
+## Node Changes (Optional Variant)
+
+A new field of `EthashVersion` defined as an 8 bit unsigned enumeration is added to 
+the the block header. If this field is absent, its value is assumed to be equal to
+ zero.
+
+The enumeration should be forward compatible and defined as:
+`EthashUndefined = 0x00  
+EthashVersion1 = 0x01  
+EthashVersion2 = 0x02  
+...  
+EthashVersion255 = 0xFF`
+
+When this field is present and set to 0x02, the `mine` and `VerifySeal` operations
+utilize the `ethashV2` algorithm. If this field is set to any value other than 0x01,
+0x02, or 0x00 `VerifySeal` shall reject the block.
+
+`VerifySeal` shall also fail verification in the event 
+`block.Number >= ASIC_MITIGATION_FORK_BLKNUM && block.EthashVersion < EthashVersion2`
+
+## Agent Changes (Optional Variant, stand alone or combined with Node Changes)
+ 
+GetWork may optionally return the proposed blocks `EthashVersion` field. While a 
```

Comment:
It can be an Algorithm field in the block, however GetWork returns unnamed values and parity already defines a 4th (block number). I would propose adding block number regardless,  and defining algorithm as an optional 5th value of getwork. Would you like version as a 6th? Or algorithm as “ethash2”?

Reply:
Since different versions of the same algorithm can't be compatible, I don't think there's any point in having separate name and version fields - just a name field should be enough.